sync/validation_queue.go

```diff
@@ -100,6 +100,11 @@ func (vq *validationQueue) traverse(s *Syncer, blk *types.BlockHeader) error {
 
 func (vq *validationQueue) finishBlockCallback(s *Syncer, block *types.Block) func() error {
 	return func() error {
+		// validate unique tx atx
+		if err := validateUniqueTxAtx(block); err != nil {
```

Comment:
do this before we validate the view so we don't validate the view of syntactically invalid blocks

Reply:
you can do it in the traverse() methode just before or after BlockSignedAndEligible check
(depends which one of them is faster)

Reply:
Done. 
@almogdepaz do you think I should add any prints there?